config/kernel-idmap_mnt_api.m4

```diff
@@ -0,0 +1,30 @@
+dnl #
+dnl # 5.12 API
+dnl #
+dnl # Check if APIs for idmapped mount are available
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_SRC_IDMAP_MNT_API], [
+        ZFS_LINUX_TEST_SRC([idmap_mnt_api], [
+                #include <linux/fs.h>
+        ],[
+		int fs_flags = FS_ALLOW_IDMAP;
+		struct user_namespace *mnt_userns = NULL;
+		struct inode *in = NULL;
+		kuid_t kid = i_uid_into_mnt(mnt_userns, in);
+		kuid_t id = kuid_from_mnt(mnt_userns, kid);
+		id = mapped_fsuid(mnt_userns);
```

Comment:
We should consider breaking this monolithic check in to one per interface.  These will change as the kernel developers see fit and if we split it up now it'll be easier to adapt to that in the future.  We can still define a `HAVE_IDMAP_MNT_API` which can be used as long as all the individual checks succeed.

Reply:
@behlendorf - I have simplified this check, also defined a few inline functions for id conversion. The generic test cases available in xfstests have been manually run and passed. Please take a look.

The zloop failure seems unrelated as I was able to reproduce it using master branch.